File: QE-SELINIUM JAVA/demo/src/main/java/demos/Activity14.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity14 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        try {
            driver.get("https://training-support.net/webelements/tables");
            System.out.println("Page Title: " + driver.getTitle());
            List<WebElement> rows = driver.findElements(
                    By.xpath("//table/tbody/tr")
            );
            List<WebElement> columns = driver.findElements(
                    By.xpath("//table/thead/tr/th")
            );

            System.out.println("Rows: " + rows.size());
            System.out.println("Columns: " + columns.size());
            WebElement bookBeforeSort = driver.findElement(
                    By.xpath("//table/tbody/tr[5]/td[2]")
            );
            
            System.out.println("Book Name (5th row before sort): "
                    + bookBeforeSort.getText());

            driver.findElement(
                    By.xpath("//table/thead/tr/th[contains(text(),'Price')]")
            ).click();
            WebElement bookAfterSort = driver.findElement(
                    By.xpath("//table/tbody/tr[5]/td[2]")
            );
            System.out.println("Book Name (5th row after sort): "
                    + bookAfterSort.getText());

        } finally {
            
            driver.quit();
        }
    }
}

File: QE-SELENIUM-python/Activity14.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()

try:
    driver.get("https://training-support.net/webelements/tables")
    print("Page Title:", driver.title)
    rows = driver.find_elements(By.XPATH, "//table/tbody/tr")
    columns = driver.find_elements(By.XPATH, "//table/thead/tr/th")

    print("Rows:", len(rows))
    print("Columns:", len(columns))
    book_before = driver.find_element(
        By.XPATH, "//table/tbody/tr[5]/td[2]"
    )
    print("Book Name (5th row before sort):", book_before.text)
    driver.find_element(
        By.XPATH, "//table/thead/tr/th[contains(text(),'Price')]"
    ).click()
    book_after = driver.find_element(
        By.XPATH, "//table/tbody/tr[5]/td[2]"
    )
    print("Book Name (5th row after sort):", book_after.text)

finally:
    driver.quit()
    